### ISSUE TITLE
spack fails to run using ubuntu 14.0.4 with system python (2.7.6)

### ISSUE THREAD

**Issue by cyrush** [LLNL/spack#22383]
### Steps to reproduce the issue

Dockerfile to reproduce:

```
FROM ubuntu:trusty

# fetch build env
RUN apt-get update && apt-get install -y \
    git \
    python \
 && rm -rf /var/lib/apt/lists/*
RUN git clone --depth 1 https://github.com/spack/spack.git
RUN cd spack && bin/spack debug report
```

### Error Message
```
------                                                                                                                                                                         
 > [4/4] RUN cd spack && bin/spack debug report:                                                                                                                               
#7 0.676 Traceback (most recent call last):                                                                                                                                    
#7 0.676   File "bin/spack", line 67, in <module>                                                                                                                              
#7 0.676     import spack.main  # noqa                                                                                                                                         
#7 0.676   File "/spack/lib/spack/spack/main.py", line 28, in <module>                                                                                                         
#7 0.676     from llnl.util.tty.log import log_output
#7 0.676   File "/spack/lib/spack/llnl/util/tty/log.py", line 23, in <module>
#7 0.676     from typing import Optional  # novm
#7 0.676 ImportError: No module named 'typing'
------
```

### Information on your system

ubuntu 14.0.4 with system python (2.7.6)

### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform (It fails in this case)
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by alalazo** [comment]
**TL;DR** This bug is "expected", since you are running Spack with Python v3.4 which is not supported.

Spack prefers for the interpreter, in order:
1. Whatever is in the `SPACK_PYTHON` environment variable
2. python3
3. python
4. python2

In your container you just pulled in a `python3` that points to a version of Python not supported by Spack.

**Comment by cyrush** [comment]
thanks for the explanation @alalazo!
